geode-web/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/ConnectCommandWithSSLMultiKeyTest.java

```diff
@@ -0,0 +1,199 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.security.SecurableCommunicationChannels;
+import org.apache.geode.test.dunit.IgnoredException;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.GfshTest;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+
+@Category(GfshTest.class)
+public class ConnectCommandWithSSLMultiKeyTest {
+  private static MemberVM locator;
+  private static Properties sslProperties;
+  private static Properties gfshSslProperties;
+  private OutputStream out = null;
+  private File sslConfigFile = null;
+
+  @Rule
+  public GfshCommandRule gfsh = new GfshCommandRule();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @ClassRule
+  public static ClusterStartupRule clusterRule = new ClusterStartupRule();
+
+  @BeforeClass
+  public static void beforeClass() {
+    // The files were generated using the following commands.
+    // The final step import the certificates in the "wrong" order (SSL implementation uses by
+    // default the first one found, ignoring the alias).
+    //
+    // openssl genrsa -out ca.key 2048
+    // # It will be necessary to provide a keystore password and the CN for each.
+    // # Use "client" for the CN of the client keystore and "server" for the CN of the server key.
+    // keytool -genkeypair -alias client -keyalg RSA -keystore client.jks
+    // keytool -genkeypair -alias server -keyalg RSA -keystore server.jks
+    // # Export CSRs
+    // keytool -certreq -alias client -keystore client.jks -file client.csr
+    // keytool -certreq -alias server -keystore server.jks -file server.csr
+    // # Create a self-signed cert for the CA
+    // openssl req -new -x509 -key ca.key -out ca.crt
+    // # Sign them
+    // openssl x509 -req -in client.csr -CA ca.crt -CAkey ca.key -out client.crt -CAcreateserial
+    // openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -out server.crt -CAcreateserial
+    // # Import both of these leaf certs into a trust store but don't import CA
+    // keytool -importcert -keystore trusted.jks -file client.crt -alias client
+    // keytool -importcert -keystore trusted.jks -file server.crt -alias server
+    // # Import the CA cert and the signed csrs back into the server and client keystores
+    // keytool -importcert -file ca.crt -keystore server.jks -storepass passw0rd -alias ca
+    // keytool -importcert -file server.crt -keystore server.jks -storepass passw0rd -alias server
+    // keytool -importcert -file ca.crt -keystore client.jks -storepass passw0rd -alias ca
+    // keytool -importcert -file client.crt -keystore client.jks -storepass passw0rd -alias client
+    String multiKeyTrustStore = createTempFileFromResource(ConnectCommandWithSSLMultiKeyTest.class,
```

Comment:
This is more of an FYI. We have a way to programmatically create keystore and truststore, but this utility is not capable of signing through CA. You can look for usages of `geode-junit/src/main/java/org/apache/geode/cache/ssl/CertStores.java`.

Reply:
Thanks for the tip @sboorlagadda, I'll have a look and see if I can use it to simplify the tests and reduce the amount of files checked in.